chain/network/src/peer/peer_actor.rs

```diff
@@ -688,12 +688,16 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
             }
             self.routed_message_cache.put(key, now);
         }
-        if let PeerMessage::Routed(RoutedMessage {
-            body: RoutedMessageBody::ForwardTx(_), ..
-        }) = &peer_msg
-        {
-            self.txns_since_last_block.fetch_add(1, Ordering::AcqRel);
-        } else if let PeerMessage::Block(_) = &peer_msg {
+        if let PeerMessage::Routed(routed) = &peer_msg {
+            match routed.as_ref() {
+                RoutedMessage { body: RoutedMessageBody::ForwardTx(_), .. } => {
+                    self.txns_since_last_block.fetch_add(1, Ordering::AcqRel);
+                }
+                _ => {
+                    self.txns_since_last_block.store(0, Ordering::Release);
```

Comment:
you changed the behavior here..


before the PR - we were 'storing' 0 only if we got a 'block' transaction. Now we're storing '0' on each non ForwardTX one.

Reply:
@mm-near You are right, thanks. That was a good catch.